app/views/schools/_form.html.erb

```diff
@@ -88,3 +99,35 @@
         class: "form-control", title: "Please enter a valid NCES ID.", id: "school_nces_id" %>
   </div>
 <%- end %>
+
+<script>
+    $(document).ready(function() {
+        const countrySelected = $('#school_country');
+        const stateSelectContainer = $('#state_select_container');
+        const stateTextfieldContainer = $('#state_textfield_container');
+        const stateSelect = $('#state_select');
+        const stateTextfield = $('#state_textfield');
+
+        function handleCountryChange() {
+            if (countrySelected.val() === 'US') {
+                stateSelectContainer.show();
+                stateSelect.attr('required', '')  //make state select required
+                stateTextfieldContainer.hide();
+
+                stateTextfield.removeAttr('name');
+                stateSelect.attr('name', 'school[state]');
+
```

Comment:
```suggestion
                stateSelectContainer.show();
                stateSelect.attr('name', 'school[state]');.attr('required', '')  // make state select required
                
                stateTextfieldContainer.hide();
                stateTextfield.removeAttr('name');
```
If there's going to be a newline, then the two types should probably be grouped together.